src/Games/NexusMods.Games.MountAndBlade2Bannerlord/LauncherManager/FileSystemProvider.cs

```diff
@@ -0,0 +1,64 @@
+using Bannerlord.LauncherManager.External;
+using Microsoft.Extensions.Logging;
+
+namespace NexusMods.Games.MountAndBlade2Bannerlord.LauncherManager;
+
+internal sealed class FileSystemProvider : IFileSystemProvider
+{
+    private readonly ILogger _logger;
+
+    public FileSystemProvider(ILogger<FileSystemProvider> logger)
+    {
+        _logger = logger;
+    }
+
+    public byte[]? ReadFileContent(string filePath, int offset, int length)
+    {
+        if (!File.Exists(filePath)) return null;
+
+        try
+        {
+            if (offset == 0 && length == -1)
+            {
+                return File.ReadAllBytes(filePath);
+            }
+            else if (offset >= 0 && length > 0)
+            {
+                var data = new byte[length];
```

Comment:
It would be preferable to use `GC.AllocateUninitializedArray<byte>(length)` here; since the array does not need to be zeroed given that we're going to immediately overwrite it anyway.